LocalizationUtility/Patches/TextTranslationPatches.cs

```diff
@@ -80,26 +69,228 @@ public static bool TextTranslation_SetLanguage(ref TextTranslation.Language lang
+                {
+                    var key = entry.Key;
+                    var value = entry.Value;
+
+                    if (language.Fixer != null) value = language.Fixer(value);
+
+                    translationTable_XML.table_ui.Add(new TextTranslation.TranslationTableEntryUI(key, value));
                 }
             }
             catch (Exception e)
             {
-                LocalizationUtility.WriteError($"Couldn't load translation for language {language.Name}: {e.Message}{e.StackTrace}");
+                LocalizationUtility.WriteError($"Couldn't add translation for language {language.Name}: {e.Message}{e.StackTrace}");
+            }
+        }
+        #endregion
+
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(TextTranslation), nameof(TextTranslation.SetLanguage))]
+        public static bool TextTranslation_SetLanguage_Prefix(ref TextTranslation.Language lang, TextTranslation __instance)
+        {
+            if (LocalizationUtility.IsVanillaLanguage(lang)) return true; //We change through TextTranslation_SetLanguage custom languages
+            //And vanilla languages with TextTranslation_SetLanguage_Postfix
+
+
+            if (!LocalizationUtility.Instance.TryGetLanguage(lang, out var language))
+            {
+                LocalizationUtility.WriteError($"The language {language} doesn't have a translation table");
                 lang = TextTranslation.Language.UNKNOWN;
                 return true;
             }
 
+            __instance.m_language = language.Language;
+            LocalizationUtility.WriteLine($"Loading translation for {language.Name}");
+            if (__instance.m_table == null) 
+            {
+                __instance.m_table.theTable = new Dictionary<string, string>();
+            }
+
+            //Now not only we can edit vanila translation tables, but multiple mods can edit the same language table, as long as the keys are unique
+            if (translationTables.TryGetValue(lang, out var table)) 
+            {
+                foreach (var pair in table.table)
+                    __instance.m_table.theTable[pair.key] = pair.value;
+
+                foreach (var pair in table.table_shipLog)
+                    __instance.m_table.theShipLogTable[pair.key] = pair.value;
+
+                foreach (var pair in table.table_ui)
+                    __instance.m_table.theUITable[pair.key] = pair.value;
+            }
+
+            var onLanguageChanged = (MulticastDelegate)__instance.GetType().GetField("OnLanguageChanged", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(__instance);
+            if (onLanguageChanged != null)
+            {
+                onLanguageChanged.DynamicInvoke();
+            }
             return false;
         }
 
+        //Maybe call this when OnLanguageChanged gets called, but could cause issues, I really don't know
+        //A *transpiler* could help to call this right after the xml is loaded (Xen would kill me)
```

Comment:
Why would I kill you tho

Reply:
Does this mean I'm allowed to make a transpiler to do that?

Reply:
idk why don't they like them? If it works without one thats fine then right?

Reply:
I guess so, well I changed the system to work with a transpiler. I think this way it is more compatible with nh and qsb as we aren't not running the original method.